part-1/5-Functions-and-beyond/README.md

```diff
@@ -1,2 +1,155 @@
-# javascript-for-testers
-A workshop series aimed at coaching testers learns JavaScript - part 1
+# Functions
+
+Function is a block of code which performs a particular task and it is reusable. Functions can be called when necessary without implementing it again.  
+
+In Javascript functions can be created in three ways
+
+* Function Declarations
+* Function Expressions
+* Arrow Functions
+
+## Function Declarations
+
+In Javascript function is created with the keyword `function` followed by function name
+
+```javascript
+function addTwoIntegers(argument1, argument2) {
+	var sum = argument1 + argument2;
+	return sum;
+}
+```
+Here the function name is `addTwoIntegers` and is  performing addition of two integers. It takes two parameters `argument1`, `argument2` and returns the output `sum`. Any code after the `return` statment will not be executed.
+
+Function will not be executed until it is called. Above function can be called by the function name.
+
+```javascript
+var number1 = 10;
+var number2 = 20;
+
+var output = addTwoIntegers(number1, number2);
+
+console.log(output); // prints 30
+```
+
+### Function Scope
+
+#### Local Variables
+Variables created inside the functions can not be accessed outside the function. Variables created inside the function have the scope limit within the function.
+
+```javascript
+function addTwoIntegers(argument1, argument2) {
+	var sum = argument1 + argument2;
+	return sum;
+}
+
+addTwoIntegers(20, 30);
+
+console.log(sum); // Uncaught ReferenceError: sum is not defined
+```
+
+#### Global Variables
+Variables defined outside the functions can be accessed inside the functions. This helps to reuse the variables.
+
+```javascript
+var sum = 0;
+
+function addTwoIntegers(argument1, argument2) {
+	sum = argument1 + argument2;
+	return sum;
+}
+
+addTwoIntegers(20, 30);
+
+console.log(sum); // prints 50
+```
+
+#### Default values for Parameters
+
+If the parameters for the functions are not passed, it will throw an error. Missing parameter is considered as `undefined`
```

Comment:
Why will it throw an error?

Reply:
Confirmed in actual project code and it will not thrown an error, missing parameter is considered as undefined. It is a typo actually.